### DISCUSSION TITLE
Can i override and change the warning message?

### DISCUSSION THREAD

**Discussion by AlexBab11** [keycloakify/keycloakify#710] [Q&A]
Hi, could you please help me to change the warning message to info and change it's text? I would like to change exactly this one, but can't find where: 
![warning](https://github.com/user-attachments/assets/19d1b770-db71-4ded-82a5-66ceba240a04)

**Comment by garronej** [comment]
Hello @AlexBab11,  

The i18n key you have to override is `verifyEmailMessage`. You must use `withCustomTranslations` or define the translation on the Keycloak server.  

Ref: https://docs.keycloakify.dev/i18n/adding-new-translation-messages-or-changing-the-default-ones

**Comment by AlexBab11** [comment]
So as you see I added the translation to this message but it wont show in storybook, is that because storybook has it's own message? Like if we add story we can see it: 
![image](https://github.com/user-attachments/assets/cb8ad6c9-808d-4246-b805-2f189dfb5fa5)
![image](https://github.com/user-attachments/assets/1e5286a7-d37f-4627-84f7-83dc6992ef99)
![image](https://github.com/user-attachments/assets/7cecba70-bf7a-4539-8961-2ca336bbd9a0)

**Reply by AlexBab11** [reply]
@garronej Yeah, i resolved the text issue but still want to ask if its possible to somehow change the type of message from warning to info?

**Comment by garronej** [comment]
> is that because storybook has it's own message? Like if we add story we can see it:  

It's not realy like that, it's just that certain messages get resolved by Keycloak (like kcContext.message.summary and kcContext.messagesPerFields).  
The storybook let's you mock the KcContext provided by the backend, you can create stories that matches your translations but Keycloakify does not do it automatically for you.  

> Yeah, i resolved the text issue but still want to ask if its possible to somehow change the type of message from warning to info?  

Sure you can change anything you want, just edit the kcContext like:  

`src/login/KcPages.tsx`
```tsx
export default function KcPage(props: { kcContext: KcContext }) {
    const { kcContext } = props;

    const { i18n } = useI18n({ kcContext });
    
    if( kcContext.pageId === "login-verify-email.ftl" ){
        kcContext.message.type = "info"
    }

    return (
        <Suspense>
            {(() => {
                switch (kcContext.pageId) {
                    default:
                        return (
                            <DefaultPage
                                kcContext={kcContext}
                                i18n={i18n}
                                classes={classes}
                                Template={Template}
                                doUseDefaultCss={true}
                                UserProfileFormFields={UserProfileFormFields}
                                doMakeUserConfirmPassword={doMakeUserConfirmPassword}
                            />
                        );
                }
            })()}
        </Suspense>
    );
}

```

**Reply by AlexBab11** [reply]
Thank you for the help!
